### PR TITLE
refactor: cssinjs for segmented

### PR DESCRIPTION
**Pull request by vagusX**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Rewrite Segmented style with cssinjs.       |
| 🇨🇳 Chinese |   重写 Segmented 样式，转而使用 cssinjs 实现。        |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by vagusX** [comment]
还缺 rtl 实现和 css logic 替换

File: components/segmented/style/index.tsx [hunk 1]
```diff
@@ -1,2 +1,210 @@
...
+    [componentCls]: {
+      ...resetComponent(token),
+
+      position: 'relative',
+      display: 'inline-flex',
+      alignItems: 'stretch',
+      justifyItems: 'flex-start',
+      color: token.segmentedLabelColor,
+      backgroundColor: token.segmentedBg,
+      borderRadius: '2px',
```
Comment (zombieJ) [inline components/segmented/style/index.tsx:73]: FIXME

File: components/segmented/style/index.tsx [hunk 2]
```diff
@@ -1,2 +1,210 @@
...
+      boxShadow: `0 0 0 2px ${token.segmentedBg}`,
```
Comment (zombieJ) [inline components/segmented/style/index.tsx:74]: FIXME

File: components/segmented/style/index.tsx [hunk 3]
```diff
@@ -1,2 +1,210 @@
...
+      transition: `all ${token.motionDurationSlow} ${token.motionEaseInOut}`,
+
+      // hover/focus styles
+      [`&:not(${componentCls}-disabled)`]: {
+        '&:hover, &:focus': {
+          backgroundColor: token.segmentedHoverBg,
+          boxShadow: `0 0 0 2px ${token.segmentedHoverBg}`,
```
Comment (zombieJ) [inline components/segmented/style/index.tsx:81]: FIXME

File: components/segmented/style/index.tsx [hunk 4]
```diff
@@ -1,2 +1,210 @@
...
+        ...segmentedDisabledItem(componentCls, token),
+      },
+
+      // thumb styles
+      [`${componentCls}-thumb`]: {
+        ...segmentedItemSelected(token),
+
+        position: 'absolute',
+        top: 0,
+        left: 0,
```
Comment (zombieJ) [inline components/segmented/style/index.tsx:165]: css logic props

**Comment by zombieJ** [comment]
https://preview-35134-ant-design.surge.sh/components/segmented-cn/#components-segmented-demo-disabled  disabled 样式丢了

**Comment by vagusX** [comment]
> [preview-35134-ant-design.surge.sh/components/segmented-cn/#components-segmented-demo-disabled](https://preview-35134-ant-design.surge.sh/components/segmented-cn/#components-segmented-demo-disabled) disabled 样式丢了

这个改好了

**Pull request merged**